src/runtime/CollectionWrappers/IterableWrapper.cs

```diff
@@ -0,0 +1,79 @@
+using System;
+using System.Collections.Generic;
+using System.Collections;
+
+namespace Python.Runtime.CollectionWrappers
+{
+    internal class IterableWrapper<T> : IEnumerable<T>
+    {
+        protected PyObject pyObject;
+
+        public IterableWrapper(PyObject pyObj)
+        {
+            pyObject = pyObj;
+        }
+
+        private void propagateIterationException()
```

Comment:
I think this will keep `StopIteration` "raised". E.g. next check for unhandled Python exception will return it.